src/main/java/crawlercommons/sitemaps/extension/ExtensionMetadata.java

```diff
@@ -36,4 +37,13 @@ public boolean isValid() {
         return true;
     }
 
+    /**
+     * Compare URLs by their string representation because calling
+     * {@link URL#equals(Object)} may trigger an unwanted and potentially slow
+     * DNS lookup to resolve the host part
+     */
+    protected static boolean urlEquals(URL a, URL b) {
+        return (a == b) || (a != null && a.toString().equals(b.toString()));
```

Comment:
if b is null, a NPE will be thrown (b.toString), can you add a null check for b also ?

Reply:
Thanks, @Chaiavi! Fixed this and same issues in the classes SitemapIndex and SiteMapURL.